lib/AST/TypeJoinMeet.cpp

```diff
@@ -271,6 +272,29 @@ CanType TypeJoin::visitExistentialMetatypeType(CanType second) {
   return ExistentialMetatypeType::get(joinInstance)->getCanonicalType();
 }
 
+CanType TypeJoin::visitExistentialType(CanType second) {
+  assert(First != second);
+
+  if (First->getKind() != second->getKind())
+    return TheAnyType;
+
+  auto firstConstraint = First->castTo<ExistentialType>()
+      ->getConstraintType()->getCanonicalType();
+  auto secondConstraint = second->castTo<ExistentialType>()
+      ->getConstraintType()->getCanonicalType();
+
+  auto joinInstance = join(firstConstraint, secondConstraint);
+  if (!joinInstance)
+    return CanType();
+
+  if (joinInstance->is<ExistentialMetatypeType>() ||
+      joinInstance->isAny() ||
+      joinInstance->isAnyObject())
```

Comment:
Note: `is<ExistentialMetatypeType>()` seems unnecessary here.

Reply:
I believe I removed this code in my follow-up PR but I'm not certain

Reply:
Ah, I did remove this bit of code in a follow-up commit in this PR. You're right that it's unnecessary.

Reply:
Whoops, sorry, I ought to have checked if it's still there on `main`.